src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -1040,38 +1040,68 @@ public class Program
     public static void Main()
     {
         object o = nameof(Main);
-        Console.WriteLine(o is string { Length is 4 });
-        Console.WriteLine(o is string { NotFound is 4 });
+        Console.WriteLine(o is P { Good is 4 });
+        Console.WriteLine(o is P { NotFound is 4 });
+        Console.WriteLine(o is P { Unreadable1 is 4 });
+        Console.WriteLine(o is P { Unreadable2 is 4 });
+
     }
 }
+class P
+{
+    public int Good = 2;
+    public int Unreadable1 { set { } }
+    public int Unreadable2 { set { } protected get { return 0; } }
+}
```

Comment:
I see a test for an inaccessible property below. Consider adding one for a field as well.